src/translate.py

```diff
@@ -1335,15 +1336,25 @@ def format(self, fmt: Formatter, force_dec: bool = False) -> str:
 
         prefix = ""
         suffix = ""
-        if not fmt.skip_casts:
+        if not fmt.skip_casts and not self.elide_cast:
             if self.type.is_pointer():
                 prefix = f"({self.type.format(fmt)})"
             if self.type.is_unsigned():
                 suffix = "U"
+
         if force_dec:
             value = str(self.value)
         else:
-            value = fmt.format_int(self.value)
+            size_bits = self.type.get_size_bits()
+            v = self.value
+            if (
+                self.type.is_signed()
+                and v > 0
```

Comment:
`and v < 2**size_bits` for safety (if it's out of range something has gone badly wrong, but we don't want to make it worse by changing literals in valid ways)